iothub/device/src/Transport/Amqp/ClientWebSocketTransport.cs

```diff
@@ -281,7 +280,7 @@ private static bool ReadTaskDone(Task<int> taskResult, TransportAsyncCallbackArg
                 args.CompletedSynchronously = ((IAsyncResult)taskResult).CompletedSynchronously;
                 return true;
             }
-            else if (taskResult.IsCanceled) // This should not happen since TaskCanceledException is handled in ReadAsyncCore.
+            else if (taskResult.IsCanceled)
```

Comment:
Not moving this to before the "completed" block?

Reply:
No, because otherwise the user wouldn't get an exception that the task was canceled.